evrMrmApp/src/drvemCML.cpp

```diff
@@ -357,31 +363,61 @@ MRMCML::getPattern(pattern p, unsigned char *buf, epicsUInt32 blen) const
     blen = std::min(plen, blen);
 
     epicsUInt32 val=0;
-    for(epicsUInt32 i=0; i<blen; i++) {
-        size_t cmlword = (i/mult);
-        size_t cmlbit  = (i%mult);
-
-        size_t cpuword, cpubit;
-        bool first; // first bit in CPU word
-
-        if(mult<32) {
-            first = cmlbit==0;
-            cpuword = cmlword;
-            cpubit = 19 - cmlbit;
-        } else {
-            first = cmlbit==0 || cmlbit==8;
-            cpuword = 2*cmlword + (cmlbit<8 ? 0 : 1);
-            cpubit  = cmlbit<8 ? 7-cmlbit : 31-(cmlbit-8);
-        }
-
-        if(first) {
-            val=shadowPattern[p][cpuword];
-        }
-
-        buf[i]=val>>cpubit;
-        buf[i]&=0x1;
-
+    
+    formFactor form = owner.getFormFactor();
+
+    if (owner.version() >=  MRFVersion(2, 7) && form == formFactor_VME64){
```

Comment:
@hongran can the last condition be converted to  be Endianness based check (`EPICS_BYTE_ORDER`) , and not platform dependent.

this check could also be implemented inside the `for` loop, where it would make it clear that only the indexes are calculated differently, but other parts are common between the different CPUs